ethereum/jsonrpc/src/main/java/tech/pegasys/pantheon/ethereum/jsonrpc/internal/methods/privacy/eea/LegacyNonceProvider.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2019 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.pantheon.ethereum.jsonrpc.internal.methods.privacy.eea;
+
+import static org.apache.logging.log4j.LogManager.getLogger;
+
+import tech.pegasys.pantheon.enclave.Enclave;
+import tech.pegasys.pantheon.enclave.types.FindPrivacyGroupRequest;
+import tech.pegasys.pantheon.enclave.types.PrivacyGroup;
+import tech.pegasys.pantheon.enclave.types.PrivacyGroup.Type;
+import tech.pegasys.pantheon.ethereum.core.Address;
+import tech.pegasys.pantheon.ethereum.privacy.PrivateTransactionHandler;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import com.google.common.collect.Lists;
+import org.apache.logging.log4j.Logger;
+import org.bouncycastle.util.Arrays;
+
+public class LegacyNonceProvider {
+
+  private static final Logger LOG = getLogger();
+
+  private final Enclave enclave;
+  private final PrivateTransactionHandler privateTransactionHandler;
+
+  public LegacyNonceProvider(
+      final Enclave enclave, final PrivateTransactionHandler privateTransactionHandler) {
+    this.enclave = enclave;
+    this.privateTransactionHandler = privateTransactionHandler;
+  }
+
+  public long determineNonce(
```

Comment:
Is there a more narrow exception to throw instead of `Exception`?

Reply:
alas - this is an effect of Encalve throwing Exception ... this behaviour is mimic'd from other Priv Json RPC ...